File: MyTestProject/Site.Models/Entities/UserProject.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;
using Site.Models.Enums;

namespace Site.Models.Entities
{
  public class UserProject : Entity<Guid>
  {
    public Guid UserId { get; set; }

    public Guid ProjectId { get; set; }

    public LinkStatus Status { get; set; }
  }
}

File: MyTestProject/Site.WEB/src/app/_models/UserProject.ts
import { User } from './user';
import { Project } from './project';

export class UserProject {
    
    userId: string;

    user: User;

    projectId: string;

    project: Project;


    constructor(userId?: string, user?: User, projectId?: string, project?: Project) {
        this.userId = userId;
        this.user = user;
        this.projectId = projectId;
        this.project = project;
    }

}
